aws/src/integration/java/org/apache/iceberg/aws/AssumeRoleAwsClientFactoryTest.java

```diff
@@ -0,0 +1,164 @@
+    iam.putRolePolicy(PutRolePolicyRequest.builder()
+        .roleName(roleName)
+        .policyName(policyName)
+        .policyDocument("{" +
+            "\"Version\":\"2012-10-17\"," +
+            "\"Statement\":[{" +
+            "\"Sid\":\"policy1\"," +
+            "\"Effect\":\"Allow\"," +
+            "\"Action\":[\"glue:CreateDatabase\",\"glue:DeleteDatabase\",\"glue:GetDatabase\",\"glue:GetTables\"]," +
+            "\"Resource\":[\"" + glueArnPrefix + ":catalog\"," +
+            "\"" + glueArnPrefix + ":database/allowed_*\"," +
+            "\"" + glueArnPrefix + ":table/allowed_*/*\"," +
+            "\"" + glueArnPrefix + ":userDefinedFunction/allowed_*/*\"]}]}")
+        .build());
+    waitForIamConsistency();
+
+    GlueCatalog glueCatalog = new GlueCatalog();
+    assumeRoleProperties.put("warehouse", "s3://path");
+    glueCatalog.initialize("test", assumeRoleProperties);
+    try {
+      glueCatalog.createNamespace(Namespace.of("denied_" + UUID.randomUUID().toString().replace("-", "")));
+      Assert.fail("Access to Glue should be denied");
+    } catch (GlueException e) {
+      Assert.assertEquals(AccessDeniedException.class, e.getClass());
+    }
+
+    try {
+      Namespace namespace = Namespace.of("allowed_" + UUID.randomUUID().toString().replace("-", ""));
+      glueCatalog.createNamespace(namespace);
+      glueCatalog.dropNamespace(namespace);
+    } catch (GlueException e) {
+      LOG.error("fail to create or delete Glue database", e);
+      Assert.fail("create and delete namespace should succeed");
+    }
+  }
+
+  @Test
+  public void testAssumeRole_s3FileIO() throws Exception {
+    String bucketArn = "arn:aws:s3:::" + AwsIntegTestUtil.testBucketName();
+    iam.putRolePolicy(PutRolePolicyRequest.builder()
+        .roleName(roleName)
+        .policyName(policyName)
+        .policyDocument("{" +
+            "\"Version\":\"2012-10-17\"," +
+            "\"Statement\":[{" +
+            "\"Sid\":\"policy1\"," +
+            "\"Effect\":\"Allow\"," +
+            "\"Action\":\"s3:ListBucket\"," +
+            "\"Resource\":[\"" + bucketArn + "\"]," +
+            "\"Condition\":{\"StringLike\":{\"s3:prefix\":[\"allowed/*\"]}}} ,{" +
+            "\"Sid\":\"policy2\"," +
+            "\"Effect\":\"Allow\"," +
+            "\"Action\":\"s3:GetObject\"," +
+            "\"Resource\":[\"" + bucketArn + "/allowed/*\"]}]}")
+        .build());
+    waitForIamConsistency();
+
+    S3FileIO s3FileIO = new S3FileIO();
+    s3FileIO.initialize(assumeRoleProperties);
+    InputFile inputFile = s3FileIO.newInputFile("s3://" + AwsIntegTestUtil.testBucketName() + "/denied/file");
+    try {
+      inputFile.exists();
+      Assert.fail("Access to s3 should be denied");
+    } catch (S3Exception e) {
+      Assert.assertEquals("Should see 403 error code", 403, e.statusCode());
+    }
+
+    inputFile = s3FileIO.newInputFile("s3://" + AwsIntegTestUtil.testBucketName() + "/allowed/file");
```

Comment:
do you delete this file?

Reply:
This file does not exist, I call `inputFile.exsits()` just to check it has access permission.